controllers/flashsystemcluster_controller.go

```diff
@@ -495,20 +502,142 @@ func (r *FlashSystemClusterReconciler) ensureExporterService(instance *odfv1alph
 		r.Log.Info("existing exporter service is expected with no change adding owner reference")
 		foundService.SetOwnerReferences(append(foundService.GetOwnerReferences(), newOwnerDetails))
 		return r.Client.Update(context.TODO(), foundService)
-
 	}
 
 	r.Log.Info("update exporter service")
 	updatedService.SetOwnerReferences(append(updatedService.GetOwnerReferences(), newOwnerDetails))
 	return r.Client.Update(context.TODO(), updatedService)
 }
 
+/*func (r *FlashSystemClusterReconciler) DeleteDuplicatedObject(instance *odfv1alpha1.FlashSystemCluster, generalObject client.ObjectList, originalKind schema.GroupVersionKind) error {
+	opts := []client.ListOption{client.InNamespace(instance.Namespace),
+		client.MatchingLabels{"odf": "storage.ibm.com"}}
+	err := r.Client.List(context.Background(), generalObject, opts...)
+	if err != nil {
+		r.Log.Error(err, "failed to list services")
+		return err
+	}
+
+	generalObject.GetObjectKind().SetGroupVersionKind(originalKind)
+	for _, currentService := range generalObject.Items {
+		if strings.Contains(currentService.Name, fsObjectsPrefix) {
+			labels := currentService.GetLabels()
+			if _, ok := labels["supportedStorage"]; !ok {
+				r.Log.Info(fmt.Sprintf("found an old FlashSystem ODF service. Deleting %v.", currentService.Name))
+				deleteService := currentService
+				err = r.Client.Delete(context.Background(), &deleteService)
+				if err != nil {
+					r.Log.Error(err, "failed to delete historical service")
+					return err
+				}
+			}
+		}
+	}
+	return nil
+}
+
+func (r *FlashSystemClusterReconciler) DeleteDuplicatedService(instance *odfv1alpha1.FlashSystemCluster) error {
+	ServicesList := &corev1.ServiceList{}
+	Groupkind := ServicesList.GetObjectKind().GroupVersionKind()
+	err := r.DeleteDuplicatedObject(instance, ServicesList, Groupkind)
+	if err != nil {
+		return err
+	}
+	return nil
+}*/
+
+func (r *FlashSystemClusterReconciler) DeleteDuplicatedService(instance *odfv1alpha1.FlashSystemCluster) error {
+	ServicesList := &corev1.ServiceList{}
+	opts := []client.ListOption{client.InNamespace(instance.Namespace),
+		client.MatchingLabels{"odf": "storage.ibm.com"}}
+	err := r.Client.List(context.Background(), ServicesList, opts...)
+	if err != nil {
+		r.Log.Error(err, "failed to list services")
+		return err
+	}
```

Comment:
Maybe if we extract this part into a new function (like `getObjectList`), it will reduce the duplicate code, and also we will use the same label filtering for all `deleteDuplicatedObjects` functions.

Reply:
added func getObjectListPerFilter